3rdparty/hal_rvv/hal_rvv_1p0/atan.hpp

```diff
@@ -0,0 +1,128 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level
+// directory of this distribution and at http://opencv.org/license.html.
+#pragma once
+
+#undef cv_hal_fastAtan32f
+#define cv_hal_fastAtan32f cv::cv_hal_rvv::fast_atan_32
+
+#undef cv_hal_fastAtan64f
+#define cv_hal_fastAtan64f cv::cv_hal_rvv::fast_atan_64
+
+#include <riscv_vector.h>
+
+#include <cfloat>
+
+namespace cv::cv_hal_rvv {
+
+namespace detail {
+// ref: mathfuncs_core.simd.hpp
```

Comment:
Could you add reference to the computation method, name, formula, whatever to identify the approach? Constants meaning and other arithmetic details will be more obvious with such details.

Reply:
This implementation didn't introduce any new math approach, the constants were simply polynomial 

https://github.com/opencv/opencv/blob/08a24ba2cf5007a2639035cc9661e18f3637223e/modules/core/src/mathfuncs_core.simd.hpp#L36-L39

Presumably it's a 7th order polynomial approximation for Arctan(x) where x in [0, 1], the optimal coefficients can be evaluated using [Remez algorithm](https://en.wikipedia.org/wiki/Remez_algorithm) I guess?

Should I add this in my code comment, or add them to 
```suggestion
// ref: mathfuncs_core.simd.hpp, 7th order polynomial form Remez algorithm? or add them to `mathfuncs_core.simd.hpp`
```